students/ScottL/session04/Mailroom_Part2.py

```diff
@@ -0,0 +1,185 @@
+    arg2 = a number containing the amount of the donation
+    """
+    print("\nDear {},\nThanks for the generous donation of ${}."
+          "\n\nSincerely,\nDonations Staff".format(name, amount))
+
+
+def add_donation(donor_name):
+    """
+    Return an appended donation list.
+    arg1 = a string containing the donor's name
+    """
+    donation_amount = input("Please enter the donation amount (or enter 'Quit' to return to "
+                            "Main Menu): ").strip("$")
+    if donation_amount.strip().lower() == "quit":
+        pass
+    else:
+        global_donation_list.append({"Name": donor_name.title(), "Amount": int(donation_amount)})
+        send_email(donor_name.title(), donation_amount)
+
+
+def receive_new_donation_info():
+    """Prompt the user to enter the donor name."""
+    user_input = input("\nPlease enter the full name of the donor, "
+                       "'List' to see a list of existing donors, "
+                       "or 'Quit' to return to the Main Menu: ")
+    if user_input.strip().lower() == "list":
+        display_donor_names()
+        receive_new_donation_info()
+    elif user_input.strip().lower() == "quit":
+        pass
+    else:
+        add_donation(user_input)
+
+
+def menu1_thank_you():
+    """Exercise the 1st option from the main menu.  Return 'True' to keep loop active."""
+    receive_new_donation_info()
+    return True
+
+
+def menu2_print_report():
+    """Exercise the 2nd option from the main menu. Return 'True' to keep loop active."""
+    display_donor_report(calc_total_donations_per_donor(),
+                         count_num_of_donations_per_donor(),
+                         calc_avg_donation_per_donor(calc_total_donations_per_donor(),
+                                                     count_num_of_donations_per_donor()))
+    return True
+
+
+def menu3_print_emails():
+    """Write 'Thank You' emails to all donors for all donations.  Return True to maintain loop"""
+    for donor in calc_total_donations_per_donor():
+        with open("{}.txt".format(donor["Name"]), "w") as file:
+            file.write("Dear {},\n\nThank you so much for your very generous donation(s) "
+                     "totaling ${}.\nWe wouldn't be able to do this without you." 
+                     "\n\nSincerely,\nExecutive Team".format(donor["Name"], donor["Amount"]))
+    return True
+
+
+def menu4_break_loop():
+    """Return a boolean False to break the loop."""
+    return False
+
+
+def main_menu():
+    """Loop through the main menu to receive and process user input."""
+    break_indicator = True
+    menu_switch = {1: menu1_thank_you,
+                   2: menu2_print_report,
+                   3: menu3_print_emails,
+                   4: menu4_break_loop}
+    while break_indicator is True:
+        display_menu_selections()
+        try:
+            user_input = int(get_user_input())
+            break_indicator = menu_switch[user_input]()
```

Comment:
this logic seems overly complex for your menu system.  You can not have to return any values for any of the functions and just keep displaying the menu system until the user selects item 4 and have that simply call `sys.exit()`

Reply:
i.e.

```
menu_switch = {1: hank_you,
                   2: print_report,
                   3: print_emails,
                   4: sys.exit()}

while True:
        display_menu_selections()
        try:
            user_input = int(get_user_input())
            menu_switch[user_input]()
        except ValueError:
            print("An invalid input was entered; please enter a valid number from the menu.")
```